src/components/AnnotationCreator.jsx

```diff
@@ -229,16 +301,49 @@ export default function AnnotationCreator({
               ),
             ];
             const theta = get(rect, 'theta', 0);
+            const newEncounterGuids = pending
+              ? []
+              : await createEncounter(
+                  sightingData,
+                  addEncounterToSighting,
+                );
+            if (addEncounterToSightingError)
+              setErrors(prevErrors => [
+                ...prevErrors,
+                addEncounterToSightingError,
+              ]);
+
+            const newEncounterGuid = get(newEncounterGuids, [0]);
+
             const newAnnotationId = await postAnnotation(
               assetId,
               IAClass,
               coords,
               viewpoint,
               theta,
+              newEncounterGuid,
             );
+            if (postAnnotationError)
+              setErrors(prevErrors => [
+                ...prevErrors,
+                postAnnotationError,
+              ]);
+
             if (newAnnotationId) {
+              if (newEncounterGuid) {
+                await addAnnotationsToSightingEncounter(
+                  newEncounterGuid,
+                  [newAnnotationId],
+                );
+                if (addToSightingEncounterError) {
+                  setErrors(prevErrors => [
+                    ...prevErrors,
+                    addToSightingEncounterError,
+                  ]);
+                }
+              }
               refreshSightingData();
-              onClose();
+              closeIfNoErrors();
```

Comment:
instead of doing this, what about looking at the response from `await addAnnotationsToSightingEncounter` and testing whether `response?.status === 200`?

Reply:
If pending is true, addAnnotationsToSightingEncounter ought never get called. So, using it's response as the deciding factor of whether the dialog should be closed would not work in the pending === true case.